### PR TITLE
First read

### PR DESCRIPTION
**Pull request by deutschbitte**

### CODE REVIEW

File: _posts/2020-09-13-state-of-swiftui.md [hunk 1]
```diff
@@ -4,38 +4,40 @@ title: "The State of SwiftUI"
...
 
-Apple released SwiftUI last year, and it's been a really exciting and wild ride. With iOS 14, a lot of the rough edges have been fixed — is SwiftUI finally ready for production?
+Apple released SwiftUI last year, and it’s been an exciting and wild ride. With iOS 14, a lot of the rough edges have been fixed — is SwiftUI finally ready for production?
 
 ## Fruta Sample App
 
-Let's look at [Apple's Fruta example](https://developer.apple.com/documentation/app_clips/fruta_building_a_feature-rich_app_with_swiftui), a cross-platform feature-rich app that's completely built in SwiftUI. It's great that Apple finally releases a more complex application for this year's cycle. I took a look when Big Sur beta 1 came out, and it's been pretty rough:
+Let’s look at [Apple’s Fruta example](https://developer.apple.com/documentation/app_clips/fruta_building_a_feature-rich_app_with_swiftui), a cross-platform feature-rich app that’s built completely in SwiftUI. It’s great that Apple is finally releasing a more complex application for this year’s cycle. 
+
+I took a look when Big Sur beta 1 came out, and it’s been pretty rough:
```
Comment (deutschbitte) [inline _posts/2020-09-13-state-of-swiftui.md:20]: What do you mean by it's been pretty rough? Is there a more specific way to phrase this than using rough? Buggy? Unrefined? Something else?

"I took a look when Big Sur beta 1 came out, and it's been pretty rough:"

File: _posts/2020-09-13-state-of-swiftui.md [hunk 2]
```diff
@@ -4,38 +4,40 @@ title: "The State of SwiftUI"
...
 
 ## SwiftUI AttributeGraph Crashes
 
-Most SwiftUI crashers are either a diffing issue in the AttributeGraph, or a bug with one of the bindings to the platform controls (AppKit or UIKit). Whenever you see `AG::Graph` in the stack trace that's Swift's AttributeGraph, which takes over representing the view hierarchy and diffing. Crashes there usually are in this form:
+Most SwiftUI crashers are either a diffing issue in AttributeGraph, or a bug with one of the bindings to the platform controls (AppKit or UIKit). Whenever you see `AG::Graph` in the stack trace, that’s Swift’s AttributeGraph, which takes over representing the view hierarchy and diffing. Crashes there usually are in this form:
```
Comment (deutschbitte) [inline _posts/2020-09-13-state-of-swiftui.md:28]: Crashers or crashes?

"Most SwiftUI crashers are either"

File: _posts/2020-09-13-state-of-swiftui.md [hunk 3]
```diff
@@ -4,38 +4,40 @@ title: "The State of SwiftUI"
...
 
-## Platform Binding Crashes
+## Platform-Binding Crashes
 
-SwiftUI uses many components from AppKit and UIKit, which is much better than to reinvent the wheel. These components are stateful and are synced with custom manager classes that perform the state diffing. These wrappers can cause issues, and as they are written in Swift there's very little possibilities to fix issues from the outside (unlike swizzling in the earlier days).
+SwiftUI uses many components from AppKit and UIKit, which is a much better strategy than reinventing the wheel. These components are stateful and are synced with custom manager classes that perform the state diffing. These wrappers can cause issues, and as they’re written in Swift, there aren’t many possibilities to fix issues from the outside (unlike with swizzling in the earlier days).
 
-Example: Removing a favorited item while it is selected crashes in the AppKit binding that syncs the SwiftUI state with `NSTableView` (FB8684522). 
+Example: Removing a favorited item while it’s selected crashes in the AppKit binding that syncs the SwiftUI state with `NSTableView` (FB8684522).
```
Comment (deutschbitte) [inline _posts/2020-09-13-state-of-swiftui.md:40]: Did you want this to link to something?

"Example: Removing a favorited item while it's selected crashes in the AppKit binding that syncs the SwiftUI state with `NSTableView` (FB8684522). "

File: _posts/2020-09-13-state-of-swiftui.md [hunk 4]
```diff
@@ -66,35 +68,35 @@ Example: Removing a favorited item while it is selected crashes in the AppKit bi
...
 ![](/assets/img/2020/fruta-swiftui/instruments.png)
 
-* Of the 10 seconds captured, 30% are used for Swift and ObjC's various retain/release and malloc's.
-* `NSAttributedString` shows up often in stack traces, giving us a hint that text layout seems especially expensive.
-* The AttributeGraph SwiftUI layout engine seems to create a lot of throwaway objects. These might mostly be Swift structs, however they still are expensive.
-* JPG decoding happens on the main thread, however is only responsible of less than 1% of the time spent here.
-* When checking "Hide System Libraries" there's basically no work done in Fruta's business logic. 
-* Sorting for "Top Functions" we see that AppKit's auto layout logic is taking up a lot of time, combined with SwiftUI's graph.
-* There seems to be a lot of unnecessary invalidation. For example, the `AppKitToolbarCoordinator` adds a toolbar item, which triggers `NSHostingView.preferencesDidChange()` causing everything to re-layout once again, even though the toolbar size doesn't change.
+* Of the 10 seconds captured, 30 percent of them are used for Swift and Objective-C’s various retain/release and malloc’s.
```
Comment (deutschbitte) [inline _posts/2020-09-13-state-of-swiftui.md:79]: What does "various retain/release and malloc's" mean? Do you mean retain/releases? Should malloc's be possessive, or plural, or? I'm not sure what to do with the last half of this sentence.

"Of the 10 seconds captured, 30 percent of them are used for Swift and Objective-C's various retain/release and malloc's."

File: _posts/2020-09-13-state-of-swiftui.md [hunk 5]
```diff
@@ -66,35 +68,35 @@ Example: Removing a favorited item while it is selected crashes in the AppKit bi
...
+@cocoawithlove proved](https://twitter.com/cocoawithlove/status/1143859576661393408) — and the UIKit port is by far faster and better than the AppKit port.
 
 ## Conclusion
 
-If your target platform is iOS 14, you're now good to go with hobby projects. I personally wouldn't yet use it for production apps, although the crash rate likely is manageable and Apple is actively improving things with every release. Remember that SwiftUI ships with the OS, not with your app, so any bug fixes will only help if your users update the OS.
+If your target platform is iOS 14, you’re now good to go with hobby projects. I personally wouldn’t yet use it for production apps, although the crash rate likely is manageable and Apple is actively improving things with every release. Remember that SwiftUI ships with the OS, not with your app, so any bug fixes will only help if your users update the OS.
```
Comment (deutschbitte) [inline _posts/2020-09-13-state-of-swiftui.md:98]: By "it" you mean SwiftUI?

"I personally wouldn't yet use it for production apps,"

File: _posts/2020-09-13-state-of-swiftui.md (hunk 1 above)
Comment (steipete) [reply _posts/2020-09-13-state-of-swiftui.md:20]: buggy; but rough feels like a more fitting term - the code just isn't refined so there are lots of small issues and problems with it. unpolished.

File: _posts/2020-09-13-state-of-swiftui.md (hunk 3 above)
Comment (steipete) [reply _posts/2020-09-13-state-of-swiftui.md:40]: no - the example is this section, stack trace is below.

File: _posts/2020-09-13-state-of-swiftui.md (hunk 4 above)
Comment (steipete) [reply _posts/2020-09-13-state-of-swiftui.md:79]: "various retain/release and malloc calls" maybe? Those are all functions and memory management terms

File: _posts/2020-09-13-state-of-swiftui.md (hunk 5 above)
Comment (steipete) [reply _posts/2020-09-13-state-of-swiftui.md:98]: yes

File: _posts/2020-09-13-state-of-swiftui.md (hunk 2 above)
Comment (steipete) [reply _posts/2020-09-13-state-of-swiftui.md:28]: crashes

**Pull request merged**